src/Spatial/Internals/ImmutableList.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace MathNet.Spatial.Internals
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    internal sealed class ImmutableList<T> : IEnumerable<T>
+    {
+        internal static readonly ImmutableList<T> Empty = new ImmutableList<T>();
+
+        private readonly T[] data;
+
+        private ImmutableList()
+        {
+            this.data = new T[0];
```

Comment:
the empty instance can be shared, not sure if it's a good idea in this case

Reply:
Not sure what can happen with an empty array. We can remove that ctor and use the other ctor and pass in an empty list, reduces noise a bit.

Reply:
there could be cases where it would be perfectly valid to return an empty list